### ISSUE TITLE
Being able to use ANY() and not just ALL() with @Auth directive for relationships

### ISSUE THREAD

**Issue by mikaelmattsson** [neo4j/graphql#1105]
```graphql
@auth(rules: [
    { where: { integration: { usersConnection: { node: { sid: "'123'" } } } } }
])
```

That creates this query:
```sql
MATCH (this:IntegrationSecret)
WHERE EXISTS((this)<-[:HAS]-(:Integration)) AND ALL(integration IN [(this)<-[:HAS]-(integration:Integration) | integration] WHERE )
RETURN this { .description } as this
```

This where input works fine when using the OGM-model or GraphQL input.

OS: MacOS
@neo4j/graphql version: 2.5.5 and 3.0.1
@neo4j/graphql-ogm version: 2.5.5 and 3.0.1
Node.js version: v16.13.2

**Comment by neo4j-team-graphql** [comment]
Many thanks for raising this bug report @mikaelmattsson. :bug: We will now attempt to reproduce the bug based on the steps you have provided.

Please ensure that you've provided the necessary information for a minimal reproduction, including but not limited to:
* Type definitions
* Resolvers
* Query and/or Mutation (or multiple) needed to reproduce

If you have a support agreement with Neo4j, please link this GitHub issue to a new or existing Zendesk ticket.

Thanks again! :pray:

**Comment by mikaelmattsson** [comment]
I have now tested som more with version 3.0.1 and tried with the new _SOME/SINGLE/ALL/NONE suffixes added to the where-relationship-inputs

Here are some the results of some of my testing:

These graphql queries work as expected
```graphql
query Secrets {
  integrationSecrets {
    description
  }
}

query Secrets {
  integrationSecrets (where: {integration: {users_SOME: {sid: "my-sid"}}}) {
    description
  }
}
```

But when I try to do it with auth it fails:
```graphql
type IntegrationSecret
@auth(rules: [
    { where: { integration: { users_SOME: { sid: "$jwt.sid" } } } },
])
{
```
graphql error message:
`"Invalid input ')': expected \"NOT\" or an expression (line 2, column 131 (offset: 161))\n\"WHERE EXISTS((this)<-[:HAS]-(:Integration)) AND ALL(integration IN [(this)<-[:HAS]-(integration:Integration) | integration] WHERE )\"\n                                                                                                                                   ^"`


This works but they way  users_All should work. 
```graphql
type IntegrationSecret
@auth(rules: [
    { where: { integration: { users: { sid: "$jwt.sid" } } } },
])
{
```

**Comment by darrellwarde** [comment]
Hey @mikaelmattsson, at the moment, the filtering syntax for `@auth` is actually different to that of queries and mutations. This is something we want to align though, and generally make other improvements, improving validation of directive definitions. So I'll relabel this as a feature request and we will treat it as such.

**Comment by darrellwarde** [comment]
No progress on this yet - it's on our radar but it's a _huge_ piece of work which will have to make up part of a major breaking release, so it's nothing we can rush into.

**Comment by darrellwarde** [comment]
We're thinking about our long-term roadmap and want to consolidate our backlog so that we can accurately plan moving forward. As such, this is being closed in favour of #1871 where we will capture all requirements in relation to filtering in the `@auth` directive.
